src/webgpu/api/validation/createTexture.spec.ts

```diff
@@ -71,58 +68,149 @@ class F extends ValidationTest {
+      .combine(
+        poptions('zeroArgument', [
+          'none',
+          'width',
+          'height',
+          'depthOrArrayLayers',
+          'mipLevelCount',
+        ] as const)
+      )
+      .combine(
+        poptions('format', [
+          'rgba8unorm',
+          'rgb10a2unorm',
+          'depth24plus-stencil8',
+          'bc1-rgba-unorm',
+        ] as const)
+      )
+  )
   .fn(async t => {
-    const format: GPUTextureFormat = t.params.format;
+    const { dimension, zeroArgument, format } = t.params;
+
+    const size = dimension === '1d' ? [31, 1, 1] : dimension === '2d' ? [31, 31, 1] : [31, 31, 31];
+    let mipLevelCount = 1;
+
+    switch (zeroArgument) {
+      case 'width':
+        size[0] = 0;
+        break;
+      case 'height':
+        size[1] = 0;
+        break;
+      case 'depthOrArrayLayers':
+        size[2] = 0;
+        break;
+      case 'mipLevelCount':
+        mipLevelCount = 0;
+        break;
+      default:
+        break;
+    }
 
     const descriptor = {
-      size: { width: 32, height: 32, depth: 1 },
-      mipLevelCount: 6,
+      size,
+      mipLevelCount,
+      dimension,
       format,
       usage: GPUTextureUsage.SAMPLED,
     };
 
-    // const extension: GPUExtensionName | undefined = kAllTextureFormatInfo[format].extension;
-    const extension: GPUExtensionName = kAllTextureFormatInfo[format].extension;
-    await t.selectDeviceOrSkipTestCase({ extensions: [extension] });
+    await t.selectDeviceOrSkipTestCase(kAllTextureFormatInfo[format].extension);
 
-    t.device.createTexture(descriptor);
+    const success = zeroArgument === 'none';
+    t.expectValidationError(() => {
+      t.device.createTexture(descriptor);
+    }, !success);
   });
 
-g.test('mipLevelCount_validation_on_dimensions')
+g.test('mipLevelCount,format')
+  .desc(
+    `Test texture creation with no mipmap chain, partial mipmap chain, full mipmap chain, out-of-bounds mipmap chain
+    for every format with different texture dimension types.`
+  )
+  .params(
+    params()
+      .combine(poptions('dimension', ['1d', '2d', '3d'] as const))
+      .combine(poptions('format', kAllTextureFormats))
+      .combine(poptions('mipLevelCount', [1, 3, 5, 6]))
+  )
+  .fn(async t => {
+    const { format, dimension, mipLevelCount } = t.params;
+    const size = dimension === '1d' ? [31, 1, 1] : dimension === '2d' ? [31, 31, 1] : [31, 31, 31];
+    const descriptor = {
+      size,
+      mipLevelCount,
+      dimension,
+      format,
+      usage: GPUTextureUsage.SAMPLED,
+    };
+
+    await t.selectDeviceOrSkipTestCase(kAllTextureFormatInfo[format].extension);
+
+    const success = mipLevelCount <= 5;
+    t.expectValidationError(() => {
+      t.device.createTexture(descriptor);
+    }, !success);
+  });
+
+g.test('mipLevelCount,bound_check')
+  .desc(
+    `Test mip level count bound check upon different texture size and different texture dimension types.`
+  )
   .params([
```

Comment:
Please switch this to either `cases` or `subcases` as `params` is deprecated. (I think it should be `subcases`.)

Reply:
Done.

Reply:
BTW, subcases will combine all sub-tests in subcases into one. We can only see much less error information. While the previous (params) will isolate each sub-test with a separate checkbox and show the error info for each sub-test.

Reply:
We intend to fix this for subcases, see "drill down" in #305, but no one is currently working on it as I don't think it's that critical.

Reply:
Got it. But it is not so convenient as before when I run the new tests I just wrote to verify whether the new tests are good. Previously, most sub-tests can pass (they are green!) if the new written CTS tests are ready for review. Right now, the whole test will fail if one single sub-test fail (that 's very common because there implementation in Chromium is not 100% ready) and I don't know whether other sub-tests can pass by first sight.